internal/subscriber/model.go

```diff
@@ -151,25 +172,63 @@ func (a *AckMessage) HasHitDeadline() bool {
 }
 
 // ParseAckID ...
-func ParseAckID(ackID string) *AckMessage {
+func ParseAckID(ackID string) (*AckMessage, error) {
 	// split the message and parse tokens
 	parts := strings.Split(ackID, ackIDSeparator)
+	decodedParts := decode(parts)
 
-	// TODO : add validations
-	partition, _ := strconv.ParseInt(utils.Decode(parts[3]), 10, 0)
-	offset, _ := strconv.ParseInt(utils.Decode(parts[4]), 10, 0)
-	deadline, _ := strconv.ParseInt(utils.Decode(parts[5]), 10, 0)
+	if !isValidAckID(decodedParts) {
+		return nil, errInvalidAckID
+	}
+
+	partition, err := strconv.ParseInt(decodedParts[3], 10, 0)
+	if err != nil {
+		err = errors.WithMessage(err, fmt.Sprintf("error in parsing partition value %v to int", decodedParts[3]))
+		return nil, err
+	}
+
+	offset, err := strconv.ParseInt(decodedParts[4], 10, 0)
+	if err != nil {
+		err = errors.WithMessage(err, fmt.Sprintf("error in parsing offset value %v to int", decodedParts[4]))
+		return nil, err
+	}
+
+	deadline, err := strconv.ParseInt(decodedParts[5], 10, 0)
+	if err != nil {
+		err = errors.WithMessage(err, fmt.Sprintf("error in parsing deadline value %v to int", decodedParts[5]))
+		return nil, err
+	}
 
 	return &AckMessage{
-		ServerAddress: utils.Decode(parts[0]),
-		SubscriberID:  utils.Decode(parts[1]),
-		Topic:         utils.Decode(parts[2]),
+		ServerAddress: decodedParts[0],
+		SubscriberID:  decodedParts[1],
+		Topic:         decodedParts[2],
 		Partition:     int32(partition),
 		Offset:        int32(offset),
 		Deadline:      int32(deadline),
-		MessageID:     utils.Decode(parts[6]),
+		MessageID:     decodedParts[6],
 		AckID:         ackID,
+	}, nil
+}
+
+// Decode the ackId parts
+func decode(parts []string) []string {
```

Comment:
move to `pkg/utils`? Technically this `func` just takes a slice of encoded strings and returns a slice of decoded strings. Nothing specific to do with this `pkg`

Reply:
make sense. Will move it to utils